mono/mini/tramp-amd64.c

```diff
@@ -363,7 +363,12 @@ mono_arch_create_generic_trampoline (MonoTrampolineType tramp_type, MonoTrampInf
 	}
 	for (i = 0; i < AMD64_XMM_NREG; ++i)
 		if (AMD64_IS_ARGUMENT_XREG (i))
+#if defined(MONO_HAVE_SIMD_REG)
 			amd64_movdqu_membase_reg (code, AMD64_RBP, saved_fpregs_offset + (i * sizeof(MonoContextSimdReg)), i);
+#else
+			/* FIXME: Is this the appropriate type on platforms without SimdReg? */
+			amd64_movdqu_membase_reg (code, AMD64_RBP, saved_fpregs_offset + (i * sizeof(mgreg_t)), i);
```

Comment:
I think you may want `amd64_movsd_membase_reg` and `sizeof(double)` as the SIMD registers are 128-bit while the other path is just storing the lower half (64-bits) into a `double`.

See code before my changes capturing the full regs here: https://github.com/NattyNarwhal/mono/commit/ce596ec25c156c754b853ea129cdb80bd6652c35#diff-e0252067fbff4f5b30094540db31db1e